sdk/storage/Azure.Storage.DataMovement/README.md

```diff
@@ -81,23 +81,51 @@ TransferManager transferManager = new TransferManager(new TransferManagerOptions
 
-A function that starts a transfer and then awaits it's completion:
+The below sample demonstrates `StorageResourceProvider` use to start transfers by uploading a file to Azure Blob Storage, using the Azure.Storage.DataMovement.Blobs package. It uses an Azure.Core token credential with permission to write to the blob.
 
 ```C# Snippet:SimpleBlobUpload_BasePackage
-async Task TransferAsync(StorageResource source, StorageResource destination,
-    TransferOptions transferOptions = default, CancellationToken cancellationToken = default)
+LocalFilesStorageResourceProvider files = new();
+BlobsStorageResourceProvider blobs = new(tokenCredential);
+DataTransfer dataTransfer = await transferManager.StartTransferAsync(
+    files.FromFile("C:/path/to/file.txt"),
+    blobs.FromBlob("https://myaccount.blob.core.windows.net/mycontainer/myblob"),
+    cancellationToken: cancellationToken);
+await dataTransfer.WaitForCompletionAsync(cancellationToken);
+```
+
+### Resuming Existing Transfers
+
+By persisting transfer progress to disk, DataMovement allows resuming of transfers that failed partway through, or were otherwise paused. To resume a transfer, the transfer manager needs to be setup in the first place with `StorageResourceProvider` instances (the same ones used above in [Starting New Transfers](#starting-new-transfers)) which are capable of reassembling the transfer components from persisted data.
+
+The below sample initializes the `TransferManager` such that it's capable of resuming transfers between the local filesystem and Azure Blob Storage, using the Azure.Storage.DataMovement.Blobs package.
+
+**Important:** Credentials to storage providers are not persisted. Storage access which requires credentials will need its appropriate `StorageResourceProvider` to be configured with those credentials. Below uses an `Azure.Core` token credential with permission to the appropriate resources.
+
+```C# Snippet:SetupTransferManagerForResume
+LocalFilesStorageResourceProvider files = new();
+BlobsStorageResourceProvider blobs = new(tokenCredential);
+TransferManager transferManager = new(new TransferManagerOptions()
 {
-    DataTransfer dataTransfer = await transferManager.StartTransferAsync(
-        source,
-        destination,
-        transferOptions,
-        cancellationToken);
-    await dataTransfer.WaitForCompletionAsync(cancellationToken);
-}
+    ResumeProviders = new List<StorageResourceProvider>() { files, blobs },
+});
 ```
 
+With a transfer manager configured as above, resuming all transfers is done as follows:
+
+```C# Snippet:ResumeAllTransfers
+List<DataTransfer> transfers = await transferManager.ResumeAllTransfersAsync();
+```
+
+An individual transfer can be resumed by transfer ID:
+
+```C# Snippet:ResumeSingleTransfer
+DataTransfer resumedTransfer = await transferManager.ResumeTransferAsync(transferId);
+```
```

Comment:
nit: Can we swap the order of resume and resume all samples? I don't want to encourage people to resume all because I think that will get them into trouble if their checkpoint directly is old/large. Currently we resumes transfers that even had one failure so that could be a lot and may not be what the customer expects. (Personally, I'm not sure we should even offer a resume all)